osu.Game/Screens/Edit/Changes/CompositeChange.cs

```diff
@@ -0,0 +1,46 @@
+// Copyright (c) ppy Pty Ltd <contact@ppy.sh>. Licensed under the MIT Licence.
+// See the LICENCE file in the repository root for full licence text.
+
+using System.Collections.Generic;
+
+namespace osu.Game.Screens.Edit.Changes
+{
+    public abstract class CompositeChange : IRevertibleChange
```

Comment:
Why does this exist at all if there is already some transaction system in place apparently?

Reply:
Seems like this is used to keep batches of similar changes in a single operation for cases like

https://github.com/ppy/osu/pull/30314/files#diff-8bdcd6dbe3a0e565f8a9720000aa6f7a1832750c269198571a5803fa72bcc8c7R13

But the question remains as to "why"? As you say, the iteration could just be done at the source instead, with the only downside potentially being higher object creation count.

Reply:
Each CompositeChange could be removed and replaced with just a static method to do the subroutine, which accepts an optional change handler.

The reason why it's here is to provide some way to do more complex editor subroutines while framing it as a `RevertibleChange` usage. It's not meant to replace transactions which are meant to represent one undo step.

I also assumed eventually we'd be able to view the change history, in which case the CompositeChange adds some interpretable grouping to the history.

If you hate this class, I can replace it with static methods that do the same operations.

Reply:
It would go a long way to have inline xmldoc/comments explaining design decisions around classes like this.